### ISSUE TITLE
Make property extraction output files have full metadata

### ISSUE THREAD

**Issue by schmie** [schmie/hemeMove#233]
**Reported by rupert on 30 Mar 2012 09:38 UTC**
In order to make the property extraction output files interpretable on their own, they must contain metadata about the geometry, similar to the snapshot files. At a minimum, this is:
* the voxel size
* the origin
You may also have to add the min/max of the bounding box too, but you'll have to check with Hywel.

Note that this must increment the extraction.h file format version number

Also, information such as the `headerSize` (!LocalPropertyOutput.cc, line 69) should be defined in `io/formats/extraction.h` so it is available to readers of the file too.

**Comment by schmie** [comment]
**Comment by miguel on 1 Jan 1970 00:22 UTC**
(In #233) Blocked on #255

**Comment by schmie** [comment]
**Comment by miguel on 1 Apr 2012 12:46 UTC**
Replying to [rupert](ticket:255):
> In order to make the property extraction output files interpretable on their own, they must contain metadata about the geometry, similar to the snapshot files. At a minimum, this is:
> * the voxel size
> * the origin

Done in e3363ccb6e61. Rupert, please CR.

> You may also have to add the min/max of the bounding box too, but you'll have to check with Hywel.

Hywel, what do you think about this?

> 
> Note that this must increment the extraction.h file format version number

Thanks for the tip. Done.

> 
> Also, information such as the `headerSize` (!LocalPropertyOutput.cc, line 69) should be defined in `io/formats/extraction.h` so it is available to readers of the file too.

I think this is not possible with the current implementation as the file header has a variable size depending on the properties requested (e.g. it includes the variable names). If this is a strong requirement, I guess we could split it into a fixed size header and a preamble with variable names, etc.

**Comment by schmie** [comment]
**Comment by miguel on 1 Apr 2012 12:47 UTC**
Hywel, please have a look at my question above.

**Comment by schmie** [comment]
**Modified by miguel on 12 Apr 2012 12:41 UTC**

**Comment by schmie** [comment]
**Comment by rupert on 19 Apr 2012 09:09 UTC**
Hywel- can you please review this for us? As I don't really understand the extraction file format, this is hard for me to review properly.
In particular, does the file contain ALL the data needed to compute the positions of the points.

Cheers

**Comment by schmie** [comment]
**Comment by hywel on 19 Apr 2012 11:06 UTC**
Extraction files are different to snapshots in this respect. Snapshots re-zero the block numbers, so that if there's no block with an x-coordinate less than 7, it subtracts 7 off all x-coords it outputs, hence you need the bounding box to recover the actual block numbers and the real-world coordinates. Extraction outputs actual block numbers so there's no need to output the bounding box.

Anyway, this all looks fine to me. And, speaking to Miguel he tested it manually so I'm happy.
